client/core/trade.go

```diff
@@ -1224,6 +1256,83 @@ func (t *trackedTrade) finalizeRedeemAction(match *matchTracker, coinID []byte)
 	return errs.ifAny()
 }
 
+func (t *trackedTrade) beginFindRedemption(match *matchTracker) {
+	swapCoinID := dex.Bytes(match.MetaData.Proof.TakerSwap)
+	fromAsset := t.wallets.fromAsset.ID
+	resultChan, err := t.wallets.fromWallet.FindRedemption(swapCoinID)
+	if err != nil {
+		log.Errorf("failed to start redemption search for match %v, %s contract %s: %v",
+			match.id, unbip(fromAsset), coinIDString(fromAsset, swapCoinID), err)
+		return
+	}
+	// Redemption search started, start goroutine to monitor result channel.
+	t.awaitingRedemption[swapCoinID.String()] = match.id
```

Comment:
Does this need a lock?

Reply:
The trade mutex is already locked before this method is called in trade.tick() (or in trade.refundMatches, which is called from trade.tick(), also with mutex locked).